controllers/osbuildconfig_controller.go

```diff
@@ -70,15 +82,137 @@ func (r *OSBuildConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
-	err = r.OSBuildCRCreator.Create(ctx, osBuildConfig)
+	newOSBuildInstanceIsNeeded, err := r.checkIfNewOSBuildInstanceIsNeeded(ctx, osBuildConfig, logger)
+	if err != nil {
+		return ctrl.Result{Requeue: true, RequeueAfter: RequeueForShortDuration}, nil
+	}
+
+	if newOSBuildInstanceIsNeeded {
+		// create new OSBuild instance
+		targetBuild := osBuildConfig.Spec.Details.TargetImage.TargetImageType
+		if targetBuild == osbuilderv1alpha1.EdgeInstallerImageType {
+			// start with edge-container OSBuild instance
+			targetBuild = osbuilderv1alpha1.EdgeContainerImageType
+		}
+		return r.createOSBuildInstance(ctx, logger, osBuildConfig, targetBuild)
+	} else {
+		return r.updateOSBuildConfigCurrentStatus(ctx, logger, osBuildConfig)
+	}
+}
+
+func (r *OSBuildConfigReconciler) checkIfNewOSBuildInstanceIsNeeded(ctx context.Context, osBuildConfig *osbuilderv1alpha1.OSBuildConfig, logger logr.Logger) (bool, error) {
+	userConfiguration := osbuilderv1alpha1.UserConfiguration{}
+	// TODO: sort before
```

Comment:
I think this `TODO` must be implemented before the change is merged. Otherwise, we're won't get a consistent behavior

Reply:
Done